framework/src/engine/generator/network_endpoint.ts

```
@@ -81,14 +81,24 @@ export class NetworkEndpoint extends BaseNetworkEndpoint {
 
 		if (Buffer.isBuffer(data)) {
 			decodedData = codec.decode<GetTransactionRequest>(getTransactionRequestSchema, data);
-			const errors = validator.validate(getTransactionRequestSchema, decodedData);
-			if (errors.length || !objectUtils.bufferArrayUniqueItems(decodedData.transactionIds)) {
-				this._logger.warn({ err: errors, peerId }, 'Received invalid getTransactions body');
+
+			try {
+				validator.validate(getTransactionRequestSchema, decodedData);
+			} catch (error) {
+				this._logger.warn({ err: error as Error, peerId }, 'Received invalid getTransactions body');
+				this.network.applyPenaltyOnPeer({
+					peerId,
+					penalty: 100,
+				});
+				throw error;
+			}
+
+			if (!objectUtils.bufferArrayUniqueItems(decodedData.transactionIds)) {
+				this._logger.warn({ peerId }, 'Received invalid getTransactions body');
 				this.network.applyPenaltyOnPeer({
 					peerId,
 					penalty: 100,
 				});
-				throw new LiskValidationError(errors);
```

in this `if` it also should throw an error, otherwise it's change the behavior